Added example

peterfriese opened the pull request:
Hi Evan,

thanks for dropbox-node, I really like it :-)

I added your example from http://evanmeagher.net/2011/01/dropbox-node-0-2-0 and added some styling. In order to make the example self-contained, I added the required modules to ./vendor. Not sure if this the best style (I'm rather new to Node.js myself), so feel free to change this.

My plan is to enhance the sample using websockets so the UI refreshes every time you make a change to the files in the currently displayed dropbox folder. I'll keep you posted as I make progress.

Again, thanks a lot and keep up the good work!

Peter

Commit:
Added example from http://evanmeagher.net/2011/01/dropbox-node-0-2-0

Signed-off-by: Peter Friese <peter.friese@itemis.de>

Commit:
Brushed up styling a bit.

Signed-off-by: Peter Friese <peter.friese@itemis.de>

Commit:
More styling.

Signed-off-by: Peter Friese <peter.friese@itemis.de>

Commit:
Third-party libs.

Signed-off-by: Peter Friese <peter.friese@itemis.de>

Commit:
Merge branch 'master' of github.com:peterfriese/dropbox-node

Commit:
Extracted dropbox secret and key, so they can be provided on the commandline.

Signed-off-by: Peter Friese <peter.friese@itemis.de>

evnm commented:
Good idea about adding the example. Thanks!

I'm running into issues getting your rework off the ground, though. One issue had to do with [express' recent requirement of a secret passed to `express.session()`](https://github.com/visionmedia/express/issues/issue/511), but there seem to be others. Does it work for you, as is?

I'm not sure how much I like keeping dependencies in a vendor dir. For one, OAuth is a strict dependency of dropbox-node as listed in npm, so it's not necessary to bundle it. As for Express and Jade, it seems to me that it would be acceptable to require users of install them themselves. As the very least, the vendor dir shouldn't be within the module root.

Commit:
Changed dependency handling. All dependencies are expected to be installed by the user using npm.

Signed-off-by: Peter Friese <peter.friese@itemis.de>

Commit:
Merge branch 'refactor_dependencies'

Commit:
Made sure it runs with the lates versions of Express / Connect.

Signed-off-by: Peter Friese <peter.friese@itemis.de>

peterfriese commented:
Thanks for your feedback!

I changed the dependency handling so the example will use the versions of Express / Connect / OAuth the user has installed. I also removed the vendor directory completely.

A local test with node 0.2.6, express 1.0.3, connect 0.5.7 and oauth 0.8.4 runs smoothly at my end.

evnm commented:
Added in [a0d3802](https://github.com/evnm/dropbox-node/commit/a0d3802).

I took a few liberties with your changes to make the example more Express-ish.

Thanks again!
